PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/utils/tlp.cpp

```diff
@@ -15,6 +17,28 @@ boost::dynamic_bitset<> TLP::getBitRep() {
 	return result;
 }
 
+
+/**
+ * int headerConsumption;
+	int dataConsumption;
+	TLPHeader* header;
+	Dllp::CreditType creditConsumedType;
+	boost::dynamic_bitset<> dataPayload;
+*/
+TLP* TLP::getObjRep(boost::dynamic_bitset<> bitset) {
+	TLP* tlp = new TLP();
+	int size = bitset.size();
+	boost::dynamic_bitset<> length_bitset = get_bits(bitset, size - 32, size - 23);
+	int lengthValue = length_bitset.to_ulong();
+	boost::dynamic_bitset<> payload_sub_bits = get_bits(bitset, 0,size- (lengthValue*4)-1);//totalsize - (lengthofheader*4) -1
+	tlp->dataPayload = payload_sub_bits;
+
+	boost::dynamic_bitset<> tlpHeader_sub_bits = get_bits(bitset, size - (lengthValue * 4), size-1);
+	tlp->header = TLPHeader::getObjRep(tlpHeader_sub_bits);
```

Comment:
You also need to add a condition to add the payload only if it's a write request. Read requests can have values in the length field but whatever the value, they should not have payloads